### DISCUSSION TITLE
Done some changes (anybody working on anything at all?)

### DISCUSSION THREAD

**Discussion by WaldemarH** [OpenSalamander/salamander#57] [General]
Hi guys

After a year I've decided that I'll never have time to work on Salamander if I'll be waiting for the free time. So I've started working on some things that bothered me for long time.. and also I just wanted to do some gui stuff just to get a feel for the code.

So here are two changes I'll be pushing sometimes next week (probably)... 
1. added the option to not change the color of the icons in the files list... so if you select/focus the file the color can remain the same -> 
![opensal_1](https://github.com/user-attachments/assets/70fd9043-39a8-4ea5-9759-09c0e3bfb361)

2. fixed the history icon issues
   1. it was to small -> now we have a bigger icon
   2. you could not change the color of it -> we are using active/hot colors now the
      same way as the disk size and zoom are.
      
   3. icon was moving constantly regarding how long the folder name was -> it has a fixed position now
   
![opensal_2](https://github.com/user-attachments/assets/83ee98b5-ec92-43e2-bbda-6dc22b56dd97)

3. what I still want to do before I'll push the changes:
   1. replace the other directory status bar bmp icons with svg ones
   2. add option that history gets reordered alphabetically
   3. add option that in history dropdown you can select subfolder the same way as in the status directory
      With this I'll also remove duplicate folders as with the new code both will be available with the longer option included.
      
![opensal_3](https://github.com/user-attachments/assets/482de81d-19d7-444d-8e89-a44745950939)

Anyway my assestment of the code:
1. the code is really written in C with some 'I wish' classes inthere
2. the code is not complex, but it is chaotic... the main issue is that it's unorganized. Well normally when a company come to this state they called me to do a full reorganization and clean it up completelly.
So if nobody is working on it I'll start transforming it to a proper C++ code as I don't see any other way to bring OS into the future (to have tabs, unicode,...).

Best regards
Waldemar

**Comment by melloware** [comment]
Excellent!

**Comment by tukanos** [comment]
I think you can go ahead and be the cleaning lady :) you promised few years back. 

It would be also great if builds would be generated automatically so even less technically inclined people could download it.

**Reply by melloware** [reply]
@tukanos i think that is on the list so the EXE would be available on the Releases section here on builds.  At least that was discussed :)

**Reply by WaldemarH** [reply]
Previous week was hectic and this week looks to be the same, but for next week it looks I'll continue with the OS. :)
Current plan:
1. finish with a few svg icons
2. change how history paths work
3. add to git and push it
4. check how we can build and release a version
   (I'll need a minor translation to be added, so we'll probably post this on the AS forum)

All in all we need to have a semy automatic process to release a new version.
Then I'll start recoding everything things from main down and encapsulate things into objects, so there will be only one global variables (the settings class) and everything else will be an object.

But first lets make the first baby steps.

**Reply by tukanos** [reply]
@WaldemarH  That would be great.  It would also mean that the cleanup is done only by you which would result in much better code overall (having multiple people can be an issue).

@melloware  I see that would be great for everyone.

**Comment by melloware** [comment]
@WaldemarH do you have a PR coming?  If so I suggest you keep your PRs small so they can be reviewed and accepted and we can start moving this project forward!

**Reply by WaldemarH** [reply]
Hi guys, yeah there are some delays... mainly I'm working to get a new source of money so that I would be able to work daily on Salamander. Currently it's looking good.

**Reply by tukanos** [reply]
It is always important to have a stable source of income.  I think your "cleaning" job will be always in demand.

**Reply by WaldemarH** [reply]
Yeah for the last year and a half I've been working toward additional income and first results are really good.

Now in the mean time I was also thinking that I would setup a Patreon site and maybe a few people could chip in a small amount so that I can cover some costs... I'm a self employed (for last 14 years) and if I don't work I don't have an income.. and my wife wouldn't understand that my heart rejoices when cleaning the Salamanders code.

Well step by step and asking God for some help can bring some good changes. :)

**Reply by tukanos** [reply]
> Yeah for the last year and a half I've been working toward additional income and first results are really good.
> 
> Now in the mean time I was also thinking that I would setup a Patreon site and maybe a few people could chip in a small amount so that I can cover some costs... I'm a self employed (for last 14 years) and if I don't work I don't have an income.. and my wife wouldn't understand that my heart rejoices when cleaning the Salamanders code.
> 
> Well step by step and asking God for some help can bring some good changes. :)

I think that is splendid idea for the users to chip in.  I think you don't have to go to Patreon, but you can do it here on [github sponsors](https://dev.to/n3wt0n/make-money-with-github-a-step-by-step-guide-12ob).  I think it would be also great, if we could vote with money for a feature.  If you want a e.g. Tabs or Unicode you commit that much to the cause.

I wish for you all the best.

**Reply by melloware** [reply]
Yep you could start a GitHub Sponsors link
